File: frontend/src/types/product.ts
export interface PricePlan {
  name: string
  price: number
  currency: string
  period: string
  features: string[]
  is_popular: boolean
}

export interface MarketplaceListing {
  name: string
  listing_url: string
  upvotes?: number
  reviews_count?: number
  rating?: number
  price_plans: PricePlan[]
  raw_data?: Record<string, any>
}

export interface TrafficInfo {
  visits_month?: number
  visits_growth?: number
  bounce_rate?: number
  avg_time_on_site?: number
  traffic_sources?: string
}

export interface MrrEstimate {
  mrr_low: number
  mrr_likely: number
  mrr_high: number
  confidence: number
  assumptions: string[]
  methodology: string
}

export interface Product {
  id: number
  name: string
  canonical_url: string
  description?: string
  logo_url?: string
  categories: string[]
  tags: string[]
  marketplaces: MarketplaceListing[]
  estimates?: MrrEstimate
  traffic?: TrafficInfo
  created_at: string
  updated_at: string
}

export interface ProductListResponse {
  products: Product[]
  total: number
  page: number
  per_page: number
}

File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class PricePlan(BaseModel):
    name: str
    price: float
    currency: str
    period: str  # monthly, annually, lifetime
    features: List[str]
    is_popular: bool = False

class MarketplaceListing(BaseModel):
    name: str
    listing_url: str
    upvotes: Optional[int] = 0
    reviews_count: Optional[int] = 0
    rating: Optional[float] = 0.0
    price_plans: List[PricePlan] = []
    raw_data: Optional[dict] = None

class TrafficInfo(BaseModel):
    visits_month: Optional[int] = 0
    visits_growth: Optional[float] = 0.0
    bounce_rate: Optional[float] = 0.0
    avg_time_on_site: Optional[float] = 0.0
    traffic_sources: Optional[str] = None

class MrrEstimate(BaseModel):
    mrr_low: float
    mrr_likely: float
    mrr_high: float
    confidence: float
    assumptions: List[str]
    methodology: str

class ProductBase(BaseModel):
    name: str
    canonical_url: str
    description: Optional[str] = None
    logo_url: Optional[str] = None
    categories: List[str] = []
    tags: List[str] = []

class ProductCreate(ProductBase):
    pass

class ProductUpdate(ProductBase):
    pass

class Product(ProductBase):
    id: int
    marketplaces: List[MarketplaceListing] = []
    estimates: Optional[MrrEstimate] = None
    traffic: Optional[TrafficInfo] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ProductList(ProductBase):
    id: int
    created_at: datetime

class ProductResponse(BaseModel):
    product: Product

class ProductListResponse(BaseModel):
    products: List[ProductList]
    total: int
    page: int
    per_page: int